cpp/source/deal.II/post_processor.templates.h

```diff
@@ -186,207 +193,248 @@ void SuperCapacitorPostprocessor<dim>::reset(
-            fe_face_values[solid_potential].get_function_values(
-                solution, face_solid_potential_values);
-            normal_vectors = fe_face_values.get_all_normal_vectors();
-            for (unsigned int face_q_point = 0; face_q_point < n_face_q_points;
-                 ++face_q_point)
+            if (cell->face(face)->boundary_id() == cathode_boundary_id)
             {
-              this->values["current"] +=
-                  (face_solid_electrical_conductivity_values[face_q_point] *
-                   face_solid_potential_gradients[face_q_point] *
-                   normal_vectors[face_q_point]) *
-                  fe_face_values.JxW(face_q_point);
-              this->values["voltage"] +=
-                  face_solid_potential_values[face_q_point] *
-                  fe_face_values.JxW(face_q_point);
-              this->values["surface_area"] += fe_face_values.JxW(face_q_point);
-            } // end for face quadrature point
-          }   // end if cathode
-        }     // end if face at boundary
-      }       // end for face
-    }         // end if cell at boundary
-  }           // end for cell
+              fe_face_values.reinit(cell, face);
+              this->mp_values->get_values(
+                  "solid_electrical_conductivity", cell,
+                  face_solid_electrical_conductivity_values); // TODO: should
+                                                              // take
+                                                              // face as an
+                                                              // argument...
+              fe_face_values[solid_potential].get_function_gradients(
+                  relevant_solution, face_solid_potential_gradients);
+              fe_face_values[solid_potential].get_function_values(
+                  relevant_solution, face_solid_potential_values);
+              normal_vectors = fe_face_values.get_all_normal_vectors();
+              for (unsigned int face_q_point = 0;
+                   face_q_point < n_face_q_points; ++face_q_point)
+              {
+                this->values["current"] +=
+                    (face_solid_electrical_conductivity_values[face_q_point] *
+                     face_solid_potential_gradients[face_q_point] *
+                     normal_vectors[face_q_point]) *
+                    fe_face_values.JxW(face_q_point);
+                this->values["voltage"] +=
+                    face_solid_potential_values[face_q_point] *
+                    fe_face_values.JxW(face_q_point);
+                this->values["surface_area"] +=
+                    fe_face_values.JxW(face_q_point);
+              } // end for face quadrature point
+            }   // end if cathode
+          }     // end if face at boundary
+        }       // end for face
+      }         // end if cell at boundary
+    }
+  } // end for cell
+  // AllReduce to get the scalar quantities
+  this->values["current"] =
```

Comment:
I think you can write something like
```
this->values["current"] = boost::mpi::all_reduce(world, this->values["current"], std::plus<double>());
```
See [here](http://www.boost.org/doc/libs/1_60_0/doc/html/boost/mpi/all_reduce.html)
That was the whole point of using Boost.MPI...